src/Symfony/Component/VarDumper/Caster/GdCaster.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\VarDumper\Caster;
+
+use Symfony\Component\VarDumper\Cloner\Stub;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ *
+ * @internal
+ */
+final class GdCaster
+{
+    public static function castGd(\GdImage $gd, array $a, Stub $stub, bool $isNested): array
+    {
+        $a['size'] = imagesx($gd).'x'.imagesy($gd);
+        $a['trueColor'] = imageistruecolor($gd);
```

Comment:
don't these miss the virtual prefix?

Reply:
Virtual prefixes weren't present until now, but I'm fine adding them, it makes sense

Reply:
they weren't present because resource's properties are always virtual
now that we deal with objects, we have to make the virtual explicitly